frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -1324,113 +1324,69 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
             },
         ],
     }),
-    urlToAction(({ actions, values, props }) => ({
+    urlToAction(({ actions, props }) => ({
         [urls.sqlEditor()]: async (_, searchParams) => {
             if (!searchParams.open_query && !searchParams.open_view && !searchParams.open_insight) {
                 return
             }
 
-            let tabAdded = false
-
-            const createQueryTab = async (): Promise<void> => {
-                if (searchParams.open_view) {
-                    // Open view
-                    const viewId = searchParams.open_view
-
-                    if (values.dataWarehouseSavedQueries.length === 0) {
-                        await dataWarehouseViewsLogic.asyncActions.loadDataWarehouseSavedQueries()
-                    }
-
-                    const view = values.dataWarehouseSavedQueries.find((n) => n.id === viewId)
-                    if (!view) {
-                        lemonToast.error('View not found')
-                        return
-                    }
-
-                    const queryToOpen = searchParams.open_query ? searchParams.open_query : view.query.query
+            // Wait for Monaco to be ready before proceeding
+            if (!props.monaco) {
```

Comment:
If this returns, it won't be triggered again so all the params handling will be lost. That's why the settimeout was there